presto-session-property-managers/src/main/java/io/prestosql/plugin/session/AbstractSessionPropertyManager.java

```diff
@@ -20,21 +20,38 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
 
 public abstract class AbstractSessionPropertyManager
         implements SessionPropertyConfigurationManager
 {
     protected abstract List<SessionMatchSpec> getSessionMatchSpecs();
 
-    @Override
-    public final Map<String, String> getSystemSessionProperties(SessionConfigurationContext context)
+    private Map<String, String> getSessionProperties(SessionConfigurationContext context)
     {
         // later properties override earlier properties
         Map<String, String> combinedProperties = new HashMap<>();
         for (SessionMatchSpec sessionMatchSpec : getSessionMatchSpecs()) {
             combinedProperties.putAll(sessionMatchSpec.match(context));
         }
-        return ImmutableMap.copyOf(combinedProperties);
+        return combinedProperties;
+    }
+
+    private static boolean isCatalogSessionProperty(Entry<String, String> property)
```

Comment:
move it to the bottom, is less interesting than public API

Reply:
Done